server/events/vcs/azuredevops_client.go

```diff
@@ -312,19 +334,14 @@ func (g *AzureDevopsClient) MergePull(pull models.PullRequest) error {
 		TriggeredByAutoComplete: new(bool),
 	}
 
-	// Construct request body from supplied parameters
-	mergePull := new(azuredevops.GitPullRequest)
-	mergePull.AutoCompleteSetBy = &id
-	mergePull.CompletionOptions = &completionOpts
-
 	owner, project, repoName := SplitAzureDevopsRepoFullName(pull.BaseRepo.FullName)
 	mergeResult, _, err := g.Client.PullRequests.Merge(
 		g.ctx,
 		owner,
 		project,
 		repoName,
 		pull.Num,
-		mergePull,
```

Comment:
Oh, and mergePull isn't actually used by the g.Client.PullRequests.Merge function call so we can set that to nil. If you want this could be in a separate PR, I feel that dropping the unnecessary setup of mergePull helps clarify the code.